### DISCUSSION TITLE
配置 v2ray.vmess.aead.forced 环境变量教程

### DISCUSSION THREAD

**Discussion by TunMax** [v2fly/v2ray-core#1514] [General]
## 1. 如果是systemctl配置运行的v2ray：

vim /etc/systemd/system/v2ray.service
在[Service]项最后增加这一行：
EnvironmentFile=/root/v2ray/v2_env

在v2_env文件上写上以下内容：
V2RAY_VMESS_AEAD_FORCED=false

systemctl restart v2ray 即可

## 2. 如果是nohup后台启用或手动./v2ray启动的：

vim /etc/profile
在最后一行写上，
export V2RAY_VMESS_AEAD_FORCED=false
退出保存。

source /etc/profile 使其即时生效。

**Comment by Tualua** [comment]
Better way will be to create an override for systemd unit and do not change the unit file itself in case it will be updated in the future

    systemctl edit v2ray.service

Add following lines

    [Service]
    Environment=V2RAY_VMESS_AEAD_FORCED=false

Restart v2ray

    systemctl restart v2ray

**Reply by DreamInvoker** [reply]
> Better way will be to create an override for systemd unit and do not change the unit file itself in case it will be updated in the future
> 
> ```
> systemctl edit v2ray.service
> ```
> 
> Add following lines
> 
> ```
> [Service]
> Environment=V2RAY_VMESS_AEAD_FORCED=false
> ```
> 
> Restart v2ray
> 
> ```
> systemctl restart v2ray
> ```

It is the true `better way` to make the environment variable work.
